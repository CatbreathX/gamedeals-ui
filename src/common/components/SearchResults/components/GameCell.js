import {
  DataItem,
  DataCell,
  DataValue, DataCellContainer,
} from 'common/components/SearchResults/components/layout/styles';
import PropTypes from 'prop-types';

export const GameCell = ({
  dealId,
  title,
  normalPrice,
  storeName,
  salePrice,
  thumbnailImage,
  priceSaving,
  metacriticRating,
  steamRatingText,
  dealRating,
}) => (
  <DataCellContainer key={dealId}>
    <DataCell>
      <img src={thumbnailImage} alt={title} />
    </DataCell>

    <DataCell>
      <DataItem className="game_cell_title">{title}</DataItem>
      <DataItem className="game_cell_store">{storeName}</DataItem>
    </DataCell>

    <DataCell>
      <DataItem>Normal Price</DataItem>
      <DataValue>{normalPrice}</DataValue>
    </DataCell>

    <DataCell>
      <DataItem>Sale Price</DataItem>
      <DataValue>{salePrice}</DataValue>
    </DataCell>

    <DataCell>
      <DataItem>Savings</DataItem>
      <DataValue>{priceSaving}</DataValue>
    </DataCell>

    <DataCell>
      <DataItem className="game_cell_metacritic-rating">{metacriticRating}</DataItem>
      <DataItem className="game_cell_steam-rating">{steamRatingText}</DataItem>
      <DataItem className="game_cell_deal-rating">{dealRating}</DataItem>
    </DataCell>

  </DataCellContainer>
);

GameCell.propTypes = {
  dealId: PropTypes.string.isRequired,
  dealRating: PropTypes.string.isRequired,
  metacriticRating: PropTypes.string.isRequired,
  normalPrice: PropTypes.string.isRequired,
  priceSaving: PropTypes.string.isRequired,
  salePrice: PropTypes.string.isRequired,
  steamRatingText: PropTypes.string.isRequired,
  storeName: PropTypes.string.isRequired,
  thumbnailImage: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
