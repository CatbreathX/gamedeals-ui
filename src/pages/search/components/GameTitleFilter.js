import { Checkbox, FormControlLabel, TextField } from '@mui/material';
import { TitleFilterContainer } from 'pages/search/layout/TitleFilterContainer';
import { Controller, useFormContext } from 'react-hook-form';

export const GameTitleFilter = () => {
  const { control, register } = useFormContext();

  return (

    <TitleFilterContainer>
      <TextField type="text" label="Game Title" {...register('gameTitle')} />
      <FormControlLabel
        control={
          <Controller
            control={control}
            name="exactGameTitle"
            render={({ field }) => {
              const isChecked = field.value === true;
              return <Checkbox {...field} checked={isChecked} />;
            }
            }
          />
        }
        label="Exact Match"
      />
    </TitleFilterContainer>
  );
};
