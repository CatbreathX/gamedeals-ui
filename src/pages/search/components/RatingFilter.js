import { FormControl, MenuItem } from '@mui/material';
import { ReactHookFormSelect } from 'common/components/inputs/ReactHookFormSelect';
import { useFormContext } from 'react-hook-form';

export const RatingFilter = () => {
  const ratings = ['50', '60', '70', '80', '85', '90', '95', '100'];

  const { control } = useFormContext();
  return (
    <FormControl sx={{
      width: 100,
    }}
    >
      <ReactHookFormSelect
        name="gameRatingFilter"
        label="Rating"
        control={control}
        defaultValue=""
      >
        <MenuItem value="">None</MenuItem>
        {ratings.map((rating) => (
          <MenuItem key={`rating${rating}`} value={rating}>
            {rating}
          </MenuItem>
        ))}
      </ReactHookFormSelect>
    </FormControl>
  );
};
