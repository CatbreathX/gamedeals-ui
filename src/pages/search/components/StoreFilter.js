import { Autocomplete, TextField } from '@mui/material';
import { sortBy } from 'lodash/collection';
import { Controller, useFormContext } from 'react-hook-form';
import { useGetStoresQuery } from 'services/gamedealapi';

export const StoreFilter = () => {
  const { control } = useFormContext();

  const {
    data,
    isSuccess,
  } = useGetStoresQuery();

  if (!isSuccess) return null;

  return (
    <>
      {isSuccess &&
        <Controller
          control={control}
          render={({ field }) => (
            <Autocomplete
              sx={{
                width: 350,
              }}
              disablePortal
              id="store-filter"
              options={createOptions(data)}
              isOptionEqualToValue={(option, value) => option.value === value.value}
              renderInput={(params) => (
                <TextField
                  {...params}
                  label="Filter Store"
                />
              )}
              onChange={(e, item) => {
                field.onChange(item?.value);
              }}
            />
          )}
          name="store"
        />
      }
      {!isSuccess &&
        <div />
      }
    </>
  );
};

const createOptions = (stores) => {
  const options = [];
  stores.forEach((store) => {
    if (store.isActive) {
      const option = {
        label: store.storeName,
        value: store.storeID,
      };

      options.push(option);
    }
  });

  const sortedOptions = sortBy(options, ['label']);
  return sortedOptions;
};
