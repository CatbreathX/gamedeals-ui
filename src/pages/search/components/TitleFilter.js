import { Checkbox, FormControlLabel, TextField } from '@mui/material';
import { TitleFilterContainer } from 'pages/search/layout/TitleFilterContainer';
import { Controller, useFormContext } from 'react-hook-form';

export const TitleFilter = () => {
  const { control, register } = useFormContext();

  return (

    <TitleFilterContainer>
      <TextField type="text" label="Title" {...register('title')} />

      <FormControlLabel
        control={
          <Controller
            control={control}
            name="exactTitle"
            render={({ field }) => {
              const isChecked = field.value === true;
              return <Checkbox {...field} checked={isChecked} />;
            }
            }
          />
        }
        label="Exact Match"
      />
    </TitleFilterContainer>
  );
};
