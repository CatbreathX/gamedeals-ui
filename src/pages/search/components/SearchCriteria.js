import { Button } from '@mui/material';
import { Form } from 'common/elements/Form';
import { H1 } from 'common/elements/H1';
import { RatingFilter } from 'pages/search/components/RatingFilter';
import { StoreFilter } from 'pages/search/components/StoreFilter';
import { TitleFilter } from 'pages/search/components/TitleFilter';
import { FilterActions, FilterContainer } from 'pages/search/layout/FilterContainer';
import PropTypes from 'prop-types';
import { FormProvider, useForm } from 'react-hook-form';

export const SearchCriteria = ({ handleFormSubmission }) => {
  const formMethods = useForm();
  const {
    handleSubmit,
  } = formMethods;

  const onFormSubmit = (data) => {
    handleFormSubmission(data);
  };

  return (
    <FormProvider {...formMethods}>
      <Form onSubmit={handleSubmit(onFormSubmit)}>
        <H1>Game Deal Search</H1>

        <FilterContainer>
          <StoreFilter />

          <RatingFilter />
          <TitleFilter />
        </FilterContainer>
        <FilterActions>
          <Button type="submit">Submit</Button>
          <Button type="button">Clear</Button>
        </FilterActions>
      </Form>
    </FormProvider>
  );
};

SearchCriteria.propTypes = {
  handleFormSubmission: PropTypes.func.isRequired,
};
