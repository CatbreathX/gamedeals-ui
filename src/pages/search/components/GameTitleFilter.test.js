import { render, screen } from '@testing-library/react';
import { GameTitleFilter } from 'pages/search/components/GameTitleFilter';
import { FormProvider, useForm } from 'react-hook-form';

describe('GameTitleFilter', () => {

  beforeEach(() => {
    render(
      <ComponentWrapper />,
    );
  });

  test('game title should be blank by default', () => {
    expect(screen.getByRole('textbox', { name: /title/i })).toHaveTextContent('');
  });

  test('should be unchecked by default', () => {
    expect(screen.getByRole('checkbox', {  name: /exact match/i})).not.toBeChecked();
  });
});

const ComponentWrapper = () => {
  const methods = useForm();

  return (
    <FormProvider {...methods}>
      <GameTitleFilter />
    </FormProvider>
  );
};
