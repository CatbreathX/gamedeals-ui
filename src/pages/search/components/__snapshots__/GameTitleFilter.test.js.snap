// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameTitleFilter should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1cdmgeq-TitleFilterContainer e18242vz0"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="mui-1"
        id="mui-1-label"
      >
        Game Title
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="mui-1"
          name="gameTitle"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1ftyaf0"
          >
            <span>
              Game Title
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          name="exactTitle"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Exact Match
      </span>
    </label>
  </div>
</DocumentFragment>
`;
